### ISSUE TITLE
Investigate wx.FileSystemHandler issue

### ISSUE THREAD

**Issue by RobinD42** [wxWidgets/Phoenix#926]
**Operating system**: Mac & Linux
**wxPython version**: 4.0.1
**Stock or custom build**: stock
**Python version**: 3.6
**Stock or custom build**: any

**Description of the problem**:

https://groups.google.com/d/msg/wxpython-dev/55_nm9Y-b7w/-tASD44PBAAJ

```python
from io import BytesIO
import wx

class MyFileSystemHandler(wx.FileSystemHandler):

    def CanOpen(self, location):
        return True

    def OpenFile(self, fs, location):
        print('OpenFile:', location)
        fsfile = wx.FSFile(BytesIO(b'hello'), location,
                          'text/plain', '', wx.DateTime.Now())
        stream = fsfile.GetStream()
        assert stream.IsOk()
        return fsfile

app = wx.App(False)
wx.FileSystem.AddHandler(MyFileSystemHandler())

hello_file = wx.FileSystem().OpenFile('myprot:hello', wx.FS_READ)
print(hello_file.GetStream().read())
```

**Comment by RobinD42** [comment]
Fixed by #1027

**Comment by bosonogi** [comment]
@RobinD42 The example works on Linux but still segfaults on Mac. I tried with wx versions 4.0.4, 4.0.6 and the 4.1.0a1.dev4234 snapshot.
